### ISSUE TITLE
Deep Representation Learning in Speech Processing: Challenges, Recent Advances, and Future Trends

### ISSUE THREAD

**Issue by hbredin** [juanmc2005/phd-bib#17]
## [Deep Representation Learning in Speech Processing: Challenges, Recent Advances, and Future Trends](https://arxiv.org/abs/2001.00378)
*Research on speech processing has traditionally considered the task of designing hand-engineered acoustic features (feature engineering) as a separate distinct problem from the task of designing efficient machine learning (ML) models to make prediction and classification decisions. There are two main drawbacks to this approach: firstly, the feature engineering being manual is cumbersome and requires human knowledge; and secondly, the designed features might not be best for the objective at hand. This has motivated the adoption of a recent trend in speech community towards utilisation of representation learning techniques, which can learn an intermediate representation of the input signal automatically that better suits the task at hand and hence lead to improved performance. The significance of representation learning has increased with advances in deep learning (DL), where the representations are more useful and less dependent on human knowledge, making it very conducive for tasks like classification, prediction, etc. The main contribution of this paper is to present an up-to-date and comprehensive survey on different techniques of speech representation learning by bringing together the scattered research across three distinct research areas including Automatic Speech Recognition (ASR), Speaker Recognition (SR), and Speaker Emotion Recognition (SER). Recent reviews in speech have been conducted for ASR, SR, and SER, however, none of these has focused on the representation learning from speech---a gap that our survey aims to bridge.*

### Cite this paper
```bibtex
@misc{latif2020deep,
    title={Deep Representation Learning in Speech Processing: Challenges, Recent Advances, and Future Trends},
    author={Siddique Latif and Rajib Rana and Sara Khalifa and Raja Jurdak and Junaid Qadir and Björn W. Schuller},
    year={2020},
    eprint={2001.00378},
    archivePrefix={arXiv},
    primaryClass={cs.SD}
}
```

**Comment by hbredin** [comment]
I did not check whether it was published somewhere.
You should probably create a tag "arXiv" so that you can clean bibtex references from time to time (by checking if papers were published)

**Comment by juanmc2005** [comment]
### Purpose
This paper is a review of the existing methods and the tendencies surrounding representation learning for speech processing.

### Architectures
Apart from the possible applications of deep representation learning, they identify the architectures that have been used in this context (MLP, CNN, RNN, Autoencoder and its variants, Boltzmann Machines and Deep Belief Networks, GAN).

### Representation Learning Techniques
Some supervised methods can be combined with unlabelled data and unsupervised models in order to take advantage of the large availability of non-annotated data, resulting in what's called **semi-supervised** training.
Among unsupervised techniques, Autoencoders and especially GANs are very popular, as they can learn representations for both inference and generation tasks.
Transfer learning methods use any data available and they have a _learn to learn_ approach. Three popular variants exist in the speech processing community:
- Domain Adaptation, which aims at reducing the mismatch between source and target domains in the representations, for instance to eliminate the difference between speaker distributions in the train and test data.
- Multi-Task Learning (MTL), which define different loss functions for the same model, thus obtaining more generalized representations. This methods however require all these types of annotations to be available in the source data.
- Self-Taught Learning is a new paradigm that combines semi-supervised learning and transfer learning. It uses labelled and unlabelled data that do not need to come from the same distribution or classes, which makes the use of different data sources easier.

Other methods are used as well, like Reinforcement Learning, Active Learning and Cooperative Learning.

### Challenges
- Challenge of training deep architectures: _"Natural speech has very complex manifolds and inherently contains information about the message, gender, age, health status, personality, friendliness, mood, and emotion. All of this information is entangled together, and the disentanglement of these attributes in some latent space is a very difficult task that requires extensive training."_ They also mention that GANs are in practice difficult to train due to vanishing gradients, convergence problems, etc.
- Performance issues of domain invariant features: _"[...] the performance of DL models drops significantly if test samples deviate from the distribution of the training data. Learning speech representations that are invariant to variabilities in speakers, language, etc., are very difficult to capture."_
- Adversary on representation learning: _" The success of adversarial attacks against DL models shows that the representations learnt by them are not good."_
- Quality of speech data can be poor mainly due to noise, bias and bad annotations.

### Recent Advances and Future Trends
- More and more toolkits and datasets are becoming open source.
- Computational advancements on GPUs, TPUs and quantum computing.
- Processing raw speech instead of using hand-crafted features.
- GANs are gaining a lot of traction for robust speech representations, speech enhancement to tackle noise and the ability to learn from synthetic data.
- Using RL to learn disentangled features by the interaction with the environment.
- Preserving privacy in the representations, as sensitive data like ethnicity or gender can be derived from them.
